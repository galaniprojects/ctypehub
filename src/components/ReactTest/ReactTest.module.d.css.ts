export const __esModule: true;
export const component: string;
