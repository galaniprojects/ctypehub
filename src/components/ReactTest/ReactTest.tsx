import { useCallback, useState } from 'react';

import * as styles from './ReactTest.module.css';

export function ReactTest() {
  const [flag, setFlag] = useState(false);
  const handleClick = useCallback(() => setFlag((old) => !old), []);
  return (
    <button onClick={handleClick} className={styles.component}>
      {flag ? '✔️' : '❌️'} Click me!
    </button>
  );
}
