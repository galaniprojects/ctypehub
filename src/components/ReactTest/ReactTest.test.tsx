import { describe, it, expect } from '@jest/globals';
import { render } from '@testing-library/react';

import { ReactTest } from './ReactTest';

describe('ReactTest', () => {
  it('should match snapshot', async () => {
    const { container } = render(<ReactTest />);
    expect(container).toMatchSnapshot();
  });
});
