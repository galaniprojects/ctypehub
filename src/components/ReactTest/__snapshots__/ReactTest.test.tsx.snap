// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactTest should match snapshot 1`] = `
<div>
  <button
    class="component"
  >
    ❌️
     Click me!
  </button>
</div>
`;
